openid-4-verifiable-presentations-1_0.md

```diff
@@ -1494,9 +1494,159 @@ issuers in Self-Sovereign Identity ecosystems using TRAIN</title>
+
+```js
+const { data } = response;
+```
+
+The Digital Credentials API offers several advantages for implementers of both Verifiers and Wallets. 
+
+Firstly, the API serves as a privacy-preserving alternative to invoking Wallets via URLs, particularly custom schemes. The user agent and app platform will only invoke a Wallet if the user confirms the request based on contextual information about the credential request and the requestor (verifier). 
+
+Secondly, the user will always be returned to the initial context, typically a browser tab, when the request has been fulfilled (or aborted), which results in an improved user experience.
+
+Thirdly, cross-device requests benefit from the use of secure transports with proximity checks, which are handled by the OS platform using FIDO CTAP 2.2 with hybrid transports.
+
+And lastly, as part of the request, the Wallet is provided with information about the Verifier's origin as authenticated by the user agent, which is important for phishing resistance.  
+
+## Protocol
+
+The value of the `protocol` parameter of the W3C Digital Credentials API MUST be set to `urn:openid.net:oid4vp` for this profile.
+
+## Request {#browser_api_request}
+
+The `request` parameter of the W3C Digital Credentials API MUST contain a valid OID4VP Authorization Request, where every Authorization Request parameter is represented as a top-level JSON member. The following is the non-normative example of such a request:
+
+```json
+{
+  "client_id": "client.example.org",
+  ...
+  "response_type": "vp_token",
+  "nonce": "n-0S6_WzA2Mj",
+  "client_metadata": {...},
+  "presentation_definition": {...}
+}
+```
+
+Out of the Authorization Request parameters defined in [@!RFC6749] and (#vp_token_request), the following are supported with this profile: 
+
+* `client_id`
+* `client_id_scheme`
+* `response_type`
+* `nonce`
+* `presentation_definition`
+* `client_metadata`
+* `request`
+* `response_mode`
+
+The `client_id` and `client_id_scheme` MUST be omitted in unsigned requests defined in (#unsigned_request). The Wallet determines the Client Identifier from the origin as asserted by the Browser. How the Wallet receives the origin from is up to the user agent and app platform and is out of scope of this profile.
+
+The value of `response_mode` parameter MUST be `w3c_dc_api` when the response is not signed and/or encrypted and `w3c_dc_api.jwt` when the response is signed and/or encrypted as defined in (#jarm).
+
+In addition to the above-mentioned parameters, this profile introduces a following new parameter:
+
+* `expected_origins`: An array of strings, each of the string representing an origin of the Verifier making the request. This parameter MUST only be used with signed requests defined in (#signed_request). It relates the logical Client Identifier to the physical endpoints that are legit origins for requests on behalf of this Client Identifier and is used to detect request replay.
```

Comment:
> It relates the logical Client Identifier to the physical endpoints that are legit origins for requests on behalf of this Client Identifier and is used to detect request replay.

Is this value specified by the verifier? What happens when it lies? Doesn't this value need to be checked against a user-agent provided value? If so, what is its purpose?

Reply:
@tplooker walked me through why a variation of this needs to be introduced and why it is useful: to prevent an interceptor re-sending this request (in a different origin) and getting the response back.

We chatted through a few options and alternatives and I think most of them would be plausible over the browser API. 

I'm going to resolve this and allow that conversation to continue between you all before taking another look at this section.

Reply:
i am not sure I follow. expected_origins only make sense in the signed request, when verifier puts expected_origins in a signed request, the wallet receives both expected_origins and the origin asserted by the browser, compares both and can detect replay. are we on the same page @samuelgoto ?